sdk/python/lib/pulumi/provider/experimental/analyzer.py

```diff
@@ -414,8 +417,30 @@ def analyze_property(
                 f"Union types are not supported: found type '{arg}' for '{typ.__name__}.{name}'"
             )
         elif is_enum(arg):
-            raise Exception(
-                f"Enum types are not supported: found type '{arg.__name__}' for '{typ.__name__}.{name}'"
+            type_name = arg.__name__
+            type_def = self.type_definitions.get(type_name)
+            if not type_def:
+                type_def = TypeDefinition(
+                    name=type_name,
+                    module=arg.__module__,
+                    type=enum_value_type(arg),
+                    properties={},
+                    properties_mapping={},
+                    description=arg.__doc__,
+                    enum=enum_members(arg),
+                    python_type=arg,
+                )
+                for member in type_def.enum or []:
+                    member.description = self.get_docstring(arg, member.name)
+                self.type_definitions[type_def.name] = type_def
+            elif type_def.module and type_def.module != arg.__module__:
+                raise DuplicateTypeError(arg.__module__, type_def)
```

Comment:
Worth reversing this if to avoid the indentation of the main block?

Reply:
I don't think that works because we still have the case where we want to skip both blocks.